app/src/main/java/com/example/gameworkflow/Point.kt

```diff
@@ -2,12 +2,17 @@ package com.example.gameworkflow
 
 data class Point(val x: Int, val y: Int)
 
-fun Point.moved(direction: Direction): Point = copy(
+private fun Point.constrainTo(bounds: Point): Point = copy(
+    x = x.coerceIn(0, bounds.x - 1),
+    y = y.coerceIn(0, bounds.y - 1)
+)
+
+private fun Point.moved(direction: Direction): Point = copy(
     x = x + direction.dx,
     y = y + direction.dy
 )
 
-fun Point.constrainTo(bounds: Point): Point = copy(
-    x = x.coerceIn(0, bounds.x - 1),
-    y = y.coerceIn(0, bounds.y - 1)
-)
+fun Point.moved(
+    direction: Direction,
+    bounds: Point
+): Point = moved(direction).constrainTo(bounds)
```

Comment:
Why this change?

Reply:
It's the only way we ever actually call it, and felt like one less bit of thinking required when reading the game workflow.